quidem-ut/src/main/java/org/apache/druid/quidem/KttmNestedComponentSupplier.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.query.NestedDataTestUtils;
+import org.apache.druid.query.QueryRunnerFactoryConglomerate;
+import org.apache.druid.segment.AutoTypeColumnSchema;
+import org.apache.druid.segment.IndexBuilder;
+import org.apache.druid.segment.IndexSpec;
+import org.apache.druid.segment.QueryableIndex;
+import org.apache.druid.segment.TestIndex;
+import org.apache.druid.segment.incremental.IncrementalIndex;
+import org.apache.druid.segment.incremental.IncrementalIndexSchema;
+import org.apache.druid.segment.join.JoinableFactoryWrapper;
+import org.apache.druid.segment.writeout.OffHeapMemorySegmentWriteOutMediumFactory;
+import org.apache.druid.server.SpecificSegmentsQuerySegmentWalker;
+import org.apache.druid.sql.calcite.TempDirProducer;
+import org.apache.druid.sql.calcite.util.SqlTestFramework.StandardComponentSupplier;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.partition.NumberedShardSpec;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public class KttmNestedComponentSupplier extends StandardComponentSupplier
+{
+  public KttmNestedComponentSupplier(TempDirProducer tempDirProducer)
+  {
+    super(tempDirProducer);
+  }
+
+  @Override
+  public SpecificSegmentsQuerySegmentWalker createQuerySegmentWalker(QueryRunnerFactoryConglomerate conglomerate,
+      JoinableFactoryWrapper joinableFactory, Injector injector)
+  {
+    SpecificSegmentsQuerySegmentWalker walker = super.createQuerySegmentWalker(conglomerate, joinableFactory, injector);
+    QueryableIndex idx = makeKttmIndex(tempDirProducer.newTempFolder());
+
+    walker.add(
+        DataSegment.builder()
+            .dataSource("kttm_nested")
+            .interval(Intervals.of("2019-08-25/2019-08-26"))
+            .version("1")
+            .shardSpec(new NumberedShardSpec(0, 0))
+            .size(0)
+            .build(),
+        idx
+    );
+    return walker;
+  }
+
+  public static QueryableIndex makeKttmIndex(File tmpDir)
+  {
+    try {
+      final File directory = new File(tmpDir, StringUtils.format("kttm-index-%s", UUID.randomUUID()));
+      final IncrementalIndex index = makeKttmNestedIndex();
+      TestIndex.INDEX_MERGER.persist(index, directory, IndexSpec.DEFAULT, null);
+      return TestIndex.INDEX_IO.loadIndex(directory);
+    }
+    catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static IncrementalIndex makeKttmNestedIndex()
+  {
+    final List<DimensionSchema> dimensions = Arrays.asList(
+        new StringDimensionSchema("session"),
+        new StringDimensionSchema("number"),
+        new AutoTypeColumnSchema("event", null),
+        new AutoTypeColumnSchema("agent", null),
+        new StringDimensionSchema("client_ip"),
+        new StringDimensionSchema("geo_ip"),
```

Comment:
this doesn't really match the way the web-console detects KTTM nested data, which calls geo_ip a json, so this should be 'auto' type schema? also language is an array in the external file, maybe it should also be 'auto' to ingest as an array since we are trying to push towards using arrays in typical use cases instead of mvds?

Reply:
updated them